test-data/unit/check-namedtuple.test

```diff
@@ -185,3 +200,104 @@ class C(B): pass
 B(1).b
 C(2).b
 [builtins fixtures/property.pyi]
+
+[case testNamedTupleAsDict]
+from collections import namedtuple, OrderedDict
+
+X = namedtuple('X', ['x', 'y'])
+x = None  # type: X
+a = None  # type: OrderedDict
+a = x._asdict()
+
+[case testNamedTupleReplace]
+from collections import namedtuple
+
+X = namedtuple('X', ['x', 'y'])
+x = None  # type: X
+x = x._replace()
+x = x._replace(y=5)
+x = x._replace(x=3)
+x = x._replace(x=3, y=5)
+x = x._replace(z=5)  # E: Unexpected keyword argument "z" for X._replace
+x = x._replace(5)  # E: Too many positional arguments for X._replace
+
+[case testNamedTupleReplaceAsClass]
+from collections import namedtuple
+
+X = namedtuple('X', ['x', 'y'])
+x = None  # type: X
+X._replace(x, x=1, y=2)
+X._replace(x=1, y=2)  # E: Missing positional argument "self" in call to X._replace
+
+[case testNamedTupleReplaceTyped]
+from typing import NamedTuple
+
+X = NamedTuple('X', [('x', int), ('y', str)])
+x = None  # type: X
+x = x._replace(x=5)
+x = x._replace(y=5)  # E: Argument 1 to X._replace has incompatible type "int"; expected "str"
+
+
+[case testNamedTupleMake]
+from typing import NamedTuple
+
+X = NamedTuple('X', [('x', int), ('y', str)])
+x = None  # type: X
+x = X._make([5, 'a'])
```

Comment:
There's no way to do that. I'd suggest using `<something>[Any]` instead, because `Union` here doesn't actually quite do the right thing, as you noticed. The `Any` makes it explicit that we can't represent this type precisely. Then you can just call it with, say, an `int` argument, and it should give an error. I think that it would be sufficient.